sql/idempotent/005-chunking.sql

```diff
@@ -0,0 +1,91 @@
+
+-------------------------------------------------------------------------------
+-- chunking_character_text_splitter
+create or replace function ai.chunking_character_text_splitter
+( chunk_column name
+, chunk_size int default 800
+, chunk_overlap int default 400
+, separator text default E'\n\n'
+, is_separator_regex bool default false
+) returns jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'character_text_splitter'
+    , 'config_type': 'chunking'
+    , 'chunk_column': chunk_column
+    , 'chunk_size': chunk_size
+    , 'chunk_overlap': chunk_overlap
+    , 'separator': separator
+    , 'is_separator_regex': is_separator_regex
+    absent on null
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- chunking_recursive_character_text_splitter
+create or replace function ai.chunking_recursive_character_text_splitter
```

Comment:
@cevian  @jgpruitt what do you think about adding a function like `ai.chunking_default(chunk_column name)` that just uses returns `ai.chunking_recursive_character_text_splitter(chunk_column)`?

Reply:
i don't think this is needed now that chunk_column is the only required arg to the chunking configs